Project/Assets/ML-Agents/Examples/GridWorld/Scripts/GridAgent.cs

```diff
@@ -105,17 +147,29 @@ public override void OnActionReceived(ActionBuffers actionBuffers)
 
             if (hit.Where(col => col.gameObject.CompareTag("goal")).ToArray().Length == 1)
             {
-                SetReward(1f);
+                ProvideReward(GridGoal.Plus);
```

Comment:
This is pretty confusing since the "goal" tag doesn't really mean that it's the goal anymore. Can you change them to e.g. "plus" and "ex"?

Or maybe this would be a good opportunity to stop using physics collision checks, and change the example to use a 2D array of enums? That would probably speed up training too.

Reply:
I changed the tags to plus and ex. I think making the grid a 2D array of enums is a good idea, but out of scope for this.